### PR TITLE
AA: implement annotation argument value

### PR DESCRIPTION
**Pull request by neetopia**
* implemented reading annotation arguments declared in Kt source.
* implemented annotation default value reading for Kt sources.
* implemented function local declarations.
* refactored `KSType.declaration` for better handling local classes.
* refactored `KSClassDeclaration` implementation to use more abstract KtSymbol.
* split annotationValue tests into kotlin and java.
* unmutes annotationValue_kt test.

### CODE REVIEW

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt [hunk 1]
```diff
@@ -36,7 +42,26 @@ class KSAnnotationImpl private constructor(private val annotationApplication: Kt
...
+                                KtNamedAnnotationValue(
+                                    valueParameterSymbol.name, KtConstantAnnotationValue(constantValue)
+                                )
+                            )
+                        }
+                    }
+                }
+            } ?: emptyList<KSValueArgument>()
+        }
+        presentArgs + absentArgs
```
Comment (ting-yuan) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt:64]: What if the args appear out of order or alternatively? E.g.,
```
annotation class A(
    val a1: Int = 1,
    val a2: Int = 2,
    val a3: Int = 3,
    val a4: Int = 4
)

@A(a3 = 1, a1 = 3)
```

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSClassDeclarationImpl.kt [hunk 2]
```diff
@@ -23,21 +23,21 @@ import org.jetbrains.kotlin.analysis.api.components.buildClassType
...
+class KSClassDeclarationImpl private constructor(private val ktClassOrObjectSymbol: KtClassOrObjectSymbol) :
     KSClassDeclaration,
-    AbstractKSDeclarationImpl(ktNamedClassOrObjectSymbol),
-    KSExpectActual by KSExpectActualImpl(ktNamedClassOrObjectSymbol) {
-    companion object : KSObjectCache<KtNamedClassOrObjectSymbol, KSClassDeclarationImpl>() {
-        fun getCached(ktNamedClassOrObjectSymbol: KtNamedClassOrObjectSymbol) =
+    AbstractKSDeclarationImpl(ktClassOrObjectSymbol),
+    KSExpectActual by KSExpectActualImpl(ktClassOrObjectSymbol) {
+    companion object : KSObjectCache<KtClassOrObjectSymbol, KSClassDeclarationImpl>() {
+        fun getCached(ktNamedClassOrObjectSymbol: KtClassOrObjectSymbol) =
```
Comment (ting-yuan) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSClassDeclarationImpl.kt:31]: nit: s/ktNamedClassOrObjectSymbol/ktClassOrObjectSymbol/

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSFunctionDeclarationImpl.kt [hunk 3]
```diff
@@ -90,7 +92,18 @@ class KSFunctionDeclarationImpl private constructor(private val ktFunctionSymbol
     }
 
     // TODO: Implement with PSI
```
Comment (ting-yuan) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSFunctionDeclarationImpl.kt:94]: Is this TODO still valid?

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt [hunk 4]
```diff
@@ -36,7 +42,26 @@ class KSAnnotationImpl private constructor(private val annotationApplication: Kt
     }
 
     override val arguments: List<KSValueArgument> by lazy {
-        annotationApplication.arguments.map { KSValueArgumentImpl.getCached(it) }
+        val presentArgs = annotationApplication.arguments.map { KSValueArgumentImpl.getCached(it) }
+        val presentNames = presentArgs.mapNotNull { it.name?.asString() }
+        val absentArgs = analyze {
+            annotationApplication.classId?.toKtClassSymbol()?.let { symbol ->
```
[the remark this answers is not in the document]
Comment (neetopia) [reply kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt:48]: This is handled in the last commit.

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt (hunk 1 above)
Comment (neetopia) [reply kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt:64]: argument is a name value pair therefore ordering is not a factor, actually this is same with FE 1.0 implementation.

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ting-yuan) [reply kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSAnnotationImpl.kt:48]: Can you elaborate? Looks like to me that it handles `isLocal` but not nested.

**Review by ting-yuan — APPROVED** [review]

**Pull request merged**